src/cluster_legacy.c

```diff
@@ -2923,8 +3033,24 @@ int clusterProcessPacket(clusterLink *link) {
     }
 
     clusterMsg *hdr = (clusterMsg *)link->rcvbuf;
-    uint16_t type = ntohs(hdr->type);
     mstime_t now = mstime();
+    int is_light = IS_LIGHT_MESSAGE(ntohs(hdr->type));
+    uint16_t type = ntohs(hdr->type) & ~CLUSTERMSG_MODIFIER_MASK;
+
+    if (is_light) {
+        if (!link->node || nodeInHandshake(link->node)) {
+            freeClusterLink(link);
+            serverLog(
+                LL_NOTICE,
+                "Closing link for node that sent a lightweight message of type %hu as its first message on the link",
+                type);
+            return 0;
+        }
+        clusterNode *sender = link->node;
+        sender->data_received = now;
+        processLightPacket(link, type);
+        return 1;
+    }
```

Comment:
This logic conceptually belongs to `processLightPacket`. `clusterProcessPacket` doesn't care when is the right time a light message should arrive. Let's move it to `processLightPacket`. Also can we consider renaming it to `clusterProcessLightPacket` to align with `clusterProcessPacket`?

```suggestion
    if (is_light) {
        clusterProcessLightPacket(link, type);
        return 1;
    }
```

Reply:
We implemented it this way so that we do not process the packets from unknown senders if publish is there first message, this maintains the `clusterProcessPacket` scheme.

Reply:
renamed `processLightPacket` to clusterProcessLightPacket and process

Reply:
>We implemented it this way so that we do not process the packets from unknown senders if publish is there first message,

This logic is fine. My point is that this logic is not `clusterProcessPacket`'s concern but `clusterProcessLightPacket`'s so let's move it into the right place?

Reply:
`clusterProcessPacket` needs to get the legit sender, only then hand it over to `clusterProcessLightPacket`'
No, IMO it is `clusterProcessPacket` logic as we also set the senders last time seen time as well here and we want to know if the sender is legit.
`sender->data_received = now;`

It is under `is_light` condition as we want to exit early and not access other hdr fields.
the above operations are done later in `clusterProcessPacket` only.

This is my perception, still if its a blocker then I can move it to the other function.

Reply:
I guess I view `clusterProcessPacket` and `clusterProcessLightPacket` as too conceptually parallel  functions at the same level. It is an implementation coincidence that `clusterProcessPacket` also plays the "dispatcher" role. So  I don't think `clusterProcessPacket` should know anything about the LIGHT message. That said, I agree we don't need to get hung out on this.